### PR TITLE
Fixed the k8s client impersonation for the authroization

### PR DESCRIPTION
**Pull request by WillardHu**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

`request.Header.Set` only one value can be set, multiple values are not supported.
```
func (h Header) Set(key, value string) {
	textproto.MIMEHeader(h).Set(key, value)
}

func (h MIMEHeader) Set(key, value string) {
	h[CanonicalMIMEHeaderKey(key)] = []string{value}
}
```

Other:
1. Make error messages more clear.
2. Add some unit tests for `RoundTrip()` function.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/common/client/impersonation.go [hunk 1]
```diff
@@ -102,21 +103,18 @@ type impersonationRoundTripper struct {
...
-			req.Header.Set(authenticationv1.ImpersonateGroupHeader, g)
+	var user, group string
+	if r.enable {
+		if v := req.Context().Value(authenticationv1.ImpersonateUserHeader); v != nil {
+			user = v.(string)
+			req.Header.Set(authenticationv1.ImpersonateUserHeader, user)
+		}
+		if v := req.Context().Value(authenticationv1.ImpersonateGroupHeader); v != nil {
+			group = v.(string)
+			req.Header[authenticationv1.ImpersonateGroupHeader] = strings.Split(group, "|")
```
Comment (fisherxu) [inline cloud/pkg/common/client/impersonation.go:114]: req.Header.Add seems can support multi values :)

**Review by fisherxu — COMMENTED** [review]
/approve

File: cloud/pkg/common/client/impersonation_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	enable bool
+}
+
+func (f *fakeNextRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if f.enable {
+		if vals := req.Header[authenticationv1.ImpersonateUserHeader]; len(vals) == 0 || vals[0] == "" {
+			return nil, fmt.Errorf("invalid request header %s", authenticationv1.ImpersonateUserHeader)
+		}
+		if vals := req.Header[authenticationv1.ImpersonateGroupHeader]; len(vals) == 0 || vals[0] == "" {
+			return nil, fmt.Errorf("invalid request header %s", authenticationv1.ImpersonateUserHeader)
```
Comment (fisherxu) [inline cloud/pkg/common/client/impersonation_test.go:26]:
```suggestion
			return nil, fmt.Errorf("invalid request header %s", authenticationv1.ImpersonateGroupHeader)
```

File: cloud/pkg/common/client/impersonation_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		}
+	} else {
+		if vals := req.Header[authenticationv1.ImpersonateUserHeader]; len(vals) > 0 {
+			return nil, fmt.Errorf("invalid request header %s", authenticationv1.ImpersonateUserHeader)
+		}
+		if vals := req.Header[authenticationv1.ImpersonateGroupHeader]; len(vals) > 0 {
+			return nil, fmt.Errorf("invalid request header %s", authenticationv1.ImpersonateUserHeader)
```
Comment (fisherxu) [inline cloud/pkg/common/client/impersonation_test.go:33]:
```suggestion
			return nil, fmt.Errorf("invalid request header %s", authenticationv1.ImpersonateGroupHeader)
```

File: cloud/pkg/common/client/impersonation_test.go (hunk 2 above)
Comment (WillardHu) [reply cloud/pkg/common/client/impersonation_test.go]: Done.

File: cloud/pkg/common/client/impersonation_test.go (hunk 3 above)
Comment (WillardHu) [reply cloud/pkg/common/client/impersonation_test.go]: Done.

**Review by fisherxu — COMMENTED** [review]
/lgtm

**Pull request merged**